### PR TITLE
Add renaming assessment models

### PR DESCRIPTION
**Pull request by bntti**
**Description of changes**
- Add support for renaming models

### CODE REVIEW

File: client/src/components/course-view/EditAssessmentModelDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
```
Comment (no-comma) [inline client/src/components/course-view/EditAssessmentModelDialog.tsx:1]:
```suggestion
// SPDX-FileCopyrightText: 2024 The Aalto Grades Developers
```

File: client/src/components/course-view/ModelsView.tsx [hunk 2]
```diff
@@ -252,19 +263,40 @@ const ModelsView = (): JSX.Element => {
                 secondaryAction={
                   editRights ? (
                     <>
-                      <IconButton
-                        onClick={() =>
-                          handleArchiveModel(model.id, !model.archived)
+                      <Tooltip placement="top" title="Edit assessment model">
```
Comment (no-comma) [inline client/src/components/course-view/ModelsView.tsx:266]: It becomes fairly obvious once you open the dialog, but should this specify that it's for editing the name specifically? The actual editing is done through the graph, right?

File: client/src/components/course-view/EditAssessmentModelDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import {useParams} from 'react-router-dom';
+
+import {useEditAssessmentModel} from '../../hooks/useApi';
+
+type EditAssessmentModelDialogProps = {
+  assessmentModelId: number | null;
+  name: string | null;
+  onClose: () => void;
+  open: boolean;
+};
```
Comment (no-comma) [inline client/src/components/course-view/EditAssessmentModelDialog.tsx:25]: Have you decided on any consistent style for writing prop types? As in, in some places they are defined as `type PropsType = {...}`, here it is defined as `type ComponentNameProps = {...}`, and in others it's just defined in the component `const ComponentName = (props: {...}) =>`

**Review by no-comma — APPROVED** [review]
LGTM, only some minor points and questions.

File: client/src/components/course-view/EditAssessmentModelDialog.tsx (hunk 3 above)
Comment (gubo97000) [reply client/src/components/course-view/EditAssessmentModelDialog.tsx:25]: Until the types are not exported, and only one component exists in a single file PropsType should be good. This case is not always respected though.

File: client/src/components/course-view/ModelsView.tsx (hunk 2 above)
Comment (gubo97000) [reply client/src/components/course-view/ModelsView.tsx:266]: I agree, "Edit name" or "Rename" would be more clear. If we add more stuff later we can call it edit info.

**Review by gubo97000 — APPROVED** [review]

**Pull request merged**